core/src/main/java/com/linecorp/armeria/server/annotation/JavaBeanRequestConverterFunction.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation;
+
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.convertParameter;
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.httpParametersOf;
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.validateAndNormalizeSupportedType;
+import static org.reflections.ReflectionUtils.getAllFields;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpParameters;
+import com.linecorp.armeria.common.util.Holder;
+import com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.EnumConverter;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.util.AsciiString;
+
+/**
+ * A default implementation of a {@link RequestConverterFunction} which converts the following parameters
+ * of the {@link AggregatedHttpMessage} to an object.
+ *
+ * <p>Path parameters to fields annotated by {@link Param}
+ *
+ * <p>HTTP parameters to fields annotated by {@link Param}
+ *
+ * <p>Headers to fields annotated by {@link Header}
+ */
+public class JavaBeanRequestConverterFunction implements RequestConverterFunction {
+    private static final ConcurrentMap<Class<?>, BeanParamInfo> beanParamInfoCache = new ConcurrentHashMap<>();
+
+    private static final List<Class<?>> ignoredResultTypeList = new ArrayList<>();
+
+    /**
+     * Initialization process for a expectedResultType.
+     *
+     * @param expectedResultType a candidate data type, which will converted by JavaBeanRequestConverterFunction
+     */
+    public static synchronized void initFor(final Class<?> expectedResultType) {
```

Comment:
I'm curious if this method could be invoked automatically in `convertRequest()` because we won't even need to expose this method in the public API then.

Reply:
We need expose it, and call it from `AnnotatedHttpServiceMethod`, to build up `BeanParamInfo` map, at the server starts up phase
see https://github.com/line/armeria/pull/1096#discussion_r174990361

Reply:
I see. How about renaming it to `register` then?

Reply:
fixed